### DISCUSSION TITLE
Ignore the ones in `.gitignore` as well

### DISCUSSION THREAD

**Discussion by smorimoto** [biomejs/biome#1016] [Ideas]
Basically, what you're adding to the `.gitignore` file is something you don't need to format. Adding them explicitly to the ignore field is a lot of effort, and ignoring them by default means you don't have to worry about synchronising them always.

Actually, Prettier does that, but ESLint doesn't... which also means that if Biome does that, it will be a brilliant ESLint! 🙂

https://github.com/prettier/prettier/blob/56408635eb01002940513e89fa5e8d9c98c9a5af/src/cli/cli-options.evaluate.js#L198

**Comment by smorimoto** [comment]
Ref: https://github.com/prettier/prettier/pull/14731

**Comment by Conaclos** [comment]
Biome supports this via the [vcs options](https://biomejs.dev/reference/configuration/#vcs).

```json
{
	"vcs": {
		"enabled": true,
		"clientKind": "git",
		"useIgnoreFile": true
	},
}
```

**Reply by smorimoto** [reply]
Sorry! I missed that! This should be promoted more and more! Fantastic!

**Reply by smorimoto** [reply]
@Conaclos Hmm, it does not handle nested `.gitignore`, right? I believe that should take care of.

**Reply by smorimoto** [reply]
Also, the first one works, but the next one does not. But both are valid ignore files.

```gitignore
dist/
```

```gitignore
/dist/
```

**Reply by ematipico** [reply]
> @Conaclos Hmm, it does not handle nested `.gitignore`, right? I believe that should take care of.

There's an issue that was filed by someone else. 

We plan to fix that, we already have an issue where we plan  to use the `ignore` crate

**Reply by smorimoto** [reply]
Fantastic!

**Reply by Psycarlo** [reply]
Awesome 🚀

**Reply by Azarattum** [reply]
I agree with [@John0x](https://github.com/biomejs/biome/discussions/1016#discussioncomment-13544206). @Conaclos, is there any reason why it couldn't be enabled by default?

**Comment by candidia** [comment]
Biome should support nested .gitignores too

**Reply by Conaclos** [reply]
We are working on the infra that could allow this.

**Comment by John0x** [comment]
I was really surprised, that it doesn't consider .gitignore by default :o
